Packages/ChatSecure/ChatSecure/Classes/Networking/Services/SubscribeAndListen/SubscribeAndListenService.swift

```diff
@@ -41,6 +42,25 @@ extension SubscribeAndListenService: ISubscribeAndListenService {
 			self.notificationListen(deviceID: deviceId, domain: server.serverDomain)
 		}
 	}
+
+	public func unSubscribe(_ server: RealmServer) {
+		guard let ownerId = server.profile?.userId else { return }
+		let deviceId = clientStore.getUniqueDeviceId()
+		var messageRequest = Message_UnSubscribeRequest()
+		messageRequest.deviceID = deviceId
+		channelStorage.getChannel(domain: server.serverDomain).unSubscribe(messageRequest) {
+			print("unSubscribe message channel success")
+			self.messageListen(deviceID: deviceId, domain: server.serverDomain, ownerId: ownerId)
+		}
+
+		var notificationRequest = Notification_UnSubscribeRequest()
+		notificationRequest.deviceID = deviceId
+		channelStorage.getChannel(domain: server.serverDomain).unSubscribe(notificationRequest) {
+			print("unSubscribe notification channel success")
+			self.notificationListen(deviceID: deviceId, domain: server.serverDomain)
+		}
+	}
+
 	
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e05ed5d402f74d55ea21b2907bb2b8dd765d1b5f/Packages/ChatSecure/ChatSecure/Classes/Networking/Services/SubscribeAndListen/SubscribeAndListenService.swift#L64"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SubscribeAndListenService.swift:64</code></td>
    </tr>
  </tbody>
</table>

Reply:
fix complete